sisl/io/siesta/ani.py

```diff
@@ -0,0 +1,61 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+"""
+Sile object for reading ANI files
+"""
+import numpy as np
+
+from .sile import SileSiesta
+from ..sile import add_sile, sile_fh_open
+from sisl._internal import set_module
+from sisl import Geometry, SuperCell
+
+__all__ = ["aniSileSiesta"]
+
+@set_module("sisl.io.siesta")
+class aniSileSiesta(SileSiesta):
```

Comment:
One could inherit from the `xyzSile` object since they are the same format. In this way you could use `super().read_geometry()`

Reply:
but we should consider adding this functionality to the `xyzSile` instead.

Reply:
> but we should consider adding this functionality to the `xyzSile` instead.

by "this functionality" you mean the possibility of returning the header?

Reply:
basically everything, an `xyz` file may be any number of entries.

Reply:
see my top comment: https://github.com/zerothi/sisl/pull/544#pullrequestreview-1321713959